### PR TITLE
Separate Cookie validation

### PR DESCRIPTION
**Pull request by JoelSpeed**
## Description

This PR does the following:
- Moves the cookie flags and defaults to their own functions
- Move CookieOption validation to its own function
- Add tests for cookie option validation
- Rename CookieOptions to Cookie

## Motivation and Context

Breaking up the validation into logical chunks

## How Has This Been Tested?

Unit tests only.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: pkg/validation/cookie.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	// Sort cookie domains by length, so that we try longer (and more specific) domains first
+	sort.Slice(o.Domains, func(i, j int) bool {
+		return len(o.Domains[i]) > len(o.Domains[j])
+	})
+
+	msgs = append(msgs, validateCookieName(o.Name)...)
+	return msgs
+}
+
+func validateCookieName(name string) []string {
```
Comment (NickMeves) [inline pkg/validation/cookie.go:37]: Should we check that the length of the cookieName is <= 256 here as well?

Comment (JoelSpeed) [reply pkg/validation/cookie.go:37]: Good idea, I'll add it

**Comment by JoelSpeed** [comment]
Added an extra commit to add your suggestion

**Comment by JoelSpeed** [comment]
Had to rebase to fix very minor conflict in changelog and `CookieOptions` to `Cookie` in 1 location, going to merge based on @NickMeves previous approval

**Pull request merged**